crates/rust-analyzer/src/config.rs

```diff
@@ -1454,6 +1457,10 @@ impl Config {
         }
     }
 
+    pub fn main_loop_num_threads(&self) -> usize {
+        self.data.numThreads.unwrap_or(num_cpus::get_physical().try_into().unwrap_or(1))
```

Comment:
If not configured, `threadpool` calls `num_cpus::get` by default, which returns the number of logical processors. So this is a change from the previous behaviour, though I think it's probably a better default.

Reply:
I chose `get_physical()` instead of `get()` because it is consistent with `prime_caches_num_threads()`. But if compatibility is an issue I think we can switch back to `get()`.

Reply:
I think it's better this way.